packages/next/src/client/components/router-reducer/reducers/fast-refresh-reducer.ts

```diff
@@ -69,11 +69,12 @@ function fastRefreshReducerImpl(
 
         // Given the path can only have two items the items are only the router state and rsc for the root.
         const [treePatch] = flightDataPath
-        const newTree = applyRouterStatePatchToTreeSkipDefault(
+        const newTree = applyRouterStatePatchToTree(
           // TODO-APP: remove ''
           [''],
           currentTree,
-          treePatch
+          treePatch,
+          location.pathname
```

Comment:
can using the global `window.location` cause any weird bugs here? like race conditions with pending navigations or something?

Reply:
Good question -- I don't think so since actions are processed sequentially - a navigation would have had to occur before this reducer would have had a chance to run.

But it might be safer to use `state.canonicalUrl` instead since that's meant to be the same value. The only reason I didn't is because it includes `location.search`, and I want to store only the `pathname` for these references. So I could probably do something like `new URL(state.canonicalUrl, location.origin).pathname` -- or maybe just `state.canonicalUrl.split('?')[0]`. Just felt kinda ugly either way 😆